### PR TITLE
T13687 debug hide vars

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: bug fix | new feature | code quality | documentation
* Link to issue: #13687 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

Thanks

### CODE REVIEW

**Comment by niden** [comment]
@sergeyklay this is ready for review. I will check appveyor in a separate branch to sync it properly with 4.0

**Comment by CameronHall** [comment]
@niden does this only cover `$_REQUEST` and `$_SERVER`? Could we also cover `$_ENV`?

**Review by sergeyklay — APPROVED** [review]
LGTM

File: ext/config.w32 [hunk 1]
```diff
@@ -13,8 +13,8 @@ if (PHP_PHALCON != "no") {
...
+	ADD_SOURCES(configure_module_dirname + "/phalcon/url", "utils.c", "phalcon");
+  ADD_SOURCES(configure_module_dirname + "/phalcon", "exception.zep.c factoryinterface.zep.c factory.zep.c dispatcherinterface.zep.c config.zep.c diinterface.zep.c flashinterface.zep.c application.zep.c di.zep.c dispatcher.zep.c flash.zep.c cryptinterface.zep.c escaperinterface.zep.c urlinterface.zep.c validationinterface.zep.c acl.zep.c container.zep.c crypt.zep.c db.zep.c debug.zep.c escaper.zep.c image.zep.c kernel.zep.c loader.zep.c logger.zep.c plugin.zep.c registry.zep.c security.zep.c tag.zep.c text.zep.c translate.zep.c url.zep.c validation.zep.c version.zep.c 0__closure.zep.c 1__closure.zep.c 2__closure.zep.c 3__closure.zep.c 4__closure.zep.c 5__closure.zep.c 6__closure.zep.c 7__closure.zep.c 8__closure.zep.c 9__closure.zep.c 10__closure.zep.c 11__closure.zep.c 12__closure.zep.c 13__closure.zep.c 14__closure.zep.c 15__closure.zep.c 16__closure.zep.c 17__closure.zep.c 18__closure.zep.c 19__closure.zep.c 20__closure.zep.c 21__closure.zep.c 22__closure.zep.c 23__closure.zep.c 24__closure.zep.c 25__closure.zep.c 26__closure.zep.c 27__closure.zep.c", "phalcon");
```
Comment (Jeckerson) [inline ext/config.w32]: More one space left ;)

File: ext/phalcon/config.zep.c [hunk 2]
```diff
@@ -409,17 +409,37 @@ PHP_METHOD(Phalcon_Config, offsetUnset) {
...
+			zephir_check_call_status();
+			break;
+		}
+		if (ZEPHIR_IS_STRING(&_0, "object")) {
+			ZEPHIR_CPY_WRT(&config, configParam);
+			break;
+		}
+		ZEPHIR_THROW_EXCEPTION_DEBUG_STR(phalcon_config_exception_ce, "Invalid data type for merge.", "phalcon/config.zep", 216);
+		return;
+	} while(0);
```
Comment (Jeckerson) [inline ext/phalcon/config.zep.c]: Why wrap inside `do while` if it won't run in cycle, isn't better to just remove it?

Comment (Jeckerson) [reply ext/phalcon/config.zep.c]: What about wrap `ZEPHIR_THROW_EXCEPTION_DEBUG_STR` inside else?

File: ext/config.w32 (hunk 1 above)
Comment (sergeyklay) [reply ext/config.w32]: It is autogenerated code  (by Zephir)

File: ext/phalcon/config.zep.c (hunk 2 above)
Comment (sergeyklay) [reply ext/phalcon/config.zep.c]: It is autogenerated code  (by Zephir)

Comment (sergeyklay) [reply ext/phalcon/config.zep.c]: > isn't better to just remove it?

Not better. It is sort of trick using C programming

File: phalcon/debug.zep [hunk 3]
```diff
@@ -331,6 +345,37 @@ class Debug
...
+			result   = [];
+
+		for value in area {
+			let value           = strtolower(value),
+				subArray[value] = 1;
+		}
+
+		let result["request"] = subArray,
+			area              = Arr::get(blacklist, "server", []),
+			subArray          = [];
```
Comment (Jeckerson) [inline phalcon/debug.zep]: Why so many spaces between variables and `=` in this method?

Comment (sergeyklay) [reply phalcon/debug.zep]: Space driven development :)

File: ext/phalcon/config.zep.c (hunk 2 above)
Comment (Jeckerson) [reply ext/phalcon/config.zep.c]: Yeah, I understood. But any way if elseif else will also work.

Comment (sergeyklay) [reply ext/phalcon/config.zep.c]: https://github.com/php/php-src/blob/PHP-7.3.0/Zend/zend_API.h#L615

Comment (sergeyklay) [reply ext/phalcon/config.zep.c]: https://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block

File: phalcon/debug.zep (hunk 3 above)
Comment (niden) [reply phalcon/debug.zep]: The tabs on the left side throw the whole thing off. Everything is actually aligned properly (equal signs and vars)

**Comment by niden** [comment]
@CameronHall The component only shows request and server, not env, this is why I did not touch that part.

File: ext/phalcon/config.zep.c (hunk 2 above)
Comment (Jeckerson) [reply ext/phalcon/config.zep.c]: OK

**Pull request merged**